src/key.rs

```diff
@@ -1602,6 +1603,36 @@ impl<'de> serde::Deserialize<'de> for XOnlyPublicKey {
     }
 }
 
+/// Sort public keys using lexicographic (of compressed serialization) order.
+/// Example:
+///
+/// ```rust
+/// # # [cfg(any(test, feature = "rand-std"))] {
+/// # use secp256k1::rand::{thread_rng, RngCore};
+/// # use secp256k1::{Secp256k1, SecretKey, Keypair, PublicKey, pubkey_sort};
+/// # let secp = Secp256k1::new();
+/// # let sk1 = SecretKey::new(&mut thread_rng());
+/// # let pub_key1 = PublicKey::from_secret_key(&secp, &sk1);
+/// # let sk2 = SecretKey::new(&mut thread_rng());
+/// # let pub_key2 = PublicKey::from_secret_key(&secp, &sk2);
+/// #
+/// # let pubkeys = [pub_key1, pub_key2];
+/// # let mut pubkeys_ref: Vec<&PublicKey> = pubkeys.iter().collect();
+/// # let pubkeys_ref = pubkeys_ref.as_mut_slice();
+/// # 
+/// # pubkey_sort(&secp, pubkeys_ref);
+/// # }
+/// ```
+pub fn pubkey_sort<C: Verification>(secp: &Secp256k1<C>, pubkeys: &mut [&PublicKey]) {
```

Comment:
In e90015e88c7604e26bc5f0dba4bc73179edc6221:

Would prefer this be called `sort_pubkeys` and be a method on `Secp256k1`. (Will do in followup.)